File: CodeGuessing/CG76/golomb.cpp
#include <unordered_map>
using namespace std;

int entry(int x)
{
    static std::unordered_map<int, int> table = {{1, 1}};
    if(table.count(x)) return table[x];

    return (table[x] = 1 + entry(x - entry(entry(x - 1))));
}

File: CodeGuessing/CG76/golomb.php
<?php
$table = array();
$table[1] = 1;

function entry($x) {
    global $table;
    if(array_key_exists($x, $table)) {
        return $table[$x];
    }
    
    $table[$x] = 1 + entry($x - entry(entry($x - 1)));
    return $table[$x];
}

?>
